src/phoenix/experimental/evals/templates/template.py

```diff
@@ -45,49 +52,45 @@ def _get_delimiters(self, delimiters: List[str]) -> Tuple[str, str]:
         else:
             raise ValueError("delimiters must only contain 2 items in the list")
 
-    def _validate(self) -> None:
+    def _validate(self, text: str) -> None:
         # Validate that for every open delimiter, we have the corresponding closing one
-        start_count = self.text.count(self._start_delim)
-        end_count = self.text.count(self._end_delim)
+        start_count = text.count(self._start_delim)
+        end_count = text.count(self._end_delim)
         if start_count != end_count:
             raise ValueError(
                 f"text poorly formatted. Found {start_count} instances of delimiter "
                 f"{self._start_delim} and {end_count} instances of {self._end_delim}. "
                 "They must be equal to be correctly paired."
             )
 
-    def _parse_variables(self) -> List[str]:
-        variables = []
-        formatter = Formatter()
-
-        text = self.text
-
-        # Example of this could be a template like: My name is ::name::
-        if self._start_delim == self._end_delim:
-            delim_length = len(self._start_delim)
-            delim_count = text.count(self._start_delim)
-            while delim_count > 0:
-                left_index = text.find(self._start_delim)
-                right_index = text[left_index + delim_length :].find(self._start_delim)
-                text = (
-                    text[0:left_index]
-                    + DEFAULT_START_DELIM
-                    + text[left_index + delim_length : left_index + delim_length + right_index]
-                    + DEFAULT_END_DELIM
-                    + text[left_index + 2 * delim_length + right_index :]
-                )
-                delim_count = text.count(self._start_delim)
-        else:
-            if self._start_delim != "{":
-                text = text.replace(self._start_delim, DEFAULT_START_DELIM)
-            if self._end_delim != "{":
-                text = text.replace(self._end_delim, DEFAULT_END_DELIM)
+    def _parse_variables(self, text: str) -> List[str]:
```

Comment:
suggestion: similar to above - doesn't need to be attached to the class I think if the function is pure.

Reply:
I agree the function is pure but to me I feel it does bake in knowledge about things like how it expects to see delimited variables in some text, so while it's not tightly coupled to any instance it is conceptually tied to the class

Reply:
I think this is where an FP person would disagree 😜 but your call ofc